src/Compilers/CSharp/Test/Semantic/IOperation/IOperationTests_IForLoopStatement.cs

```diff
@@ -0,0 +1,2041 @@
+        Right: IBinaryOperatorExpression (BinaryOperationKind.IntegerAdd) (OperationKind.BinaryOperatorExpression, Type: System.Int32)
+            Left: ILocalReferenceExpression: i (OperationKind.LocalReferenceExpression, Type: System.Int32)
+            Right: ILiteralExpression (Text: 2) (OperationKind.LiteralExpression, Type: System.Int32, Constant: 2)
+    IExpressionStatement (OperationKind.ExpressionStatement)
+      IAssignmentExpression (OperationKind.AssignmentExpression, Type: System.Int32)
+        Left: ILocalReferenceExpression: j (OperationKind.LocalReferenceExpression, Type: System.Int32)
+        Right: IBinaryOperatorExpression (BinaryOperationKind.IntegerAdd) (OperationKind.BinaryOperatorExpression, Type: System.Int32)
+            Left: ILocalReferenceExpression: j (OperationKind.LocalReferenceExpression, Type: System.Int32)
+            Right: ILiteralExpression (Text: 1) (OperationKind.LiteralExpression, Type: System.Int32, Constant: 1)
+  IBlockStatement (0 statements) (OperationKind.BlockStatement)
+";
+            VerifyOperationTreeForTest<ForStatementSyntax>(source, expectedOperationTree);
+        }
+
+        [Fact, WorkItem(17602, "https://github.com/dotnet/roslyn/issues/17602")]
+        public void IForLoopStatement_InitializerList()
+        {
+            string source = @"
+class C
+{
+    static void Main(string[] args)
+    {
+        int i = 1;
+        /*<bind>*/for (; i < 10; i = i + 1)
+        {
+        }/*</bind>*/
+    }
+}
+
+";
+            string expectedOperationTree = @"
+IForLoopStatement (LoopKind.For) (OperationKind.LoopStatement)
+  Condition: IBinaryOperatorExpression (BinaryOperationKind.IntegerLessThan) (OperationKind.BinaryOperatorExpression, Type: System.Boolean)
+      Left: ILocalReferenceExpression: i (OperationKind.LocalReferenceExpression, Type: System.Int32)
+      Right: ILiteralExpression (Text: 10) (OperationKind.LiteralExpression, Type: System.Int32, Constant: 10)
+  AtLoopBottom: IExpressionStatement (OperationKind.ExpressionStatement)
+      IAssignmentExpression (OperationKind.AssignmentExpression, Type: System.Int32)
+        Left: ILocalReferenceExpression: i (OperationKind.LocalReferenceExpression, Type: System.Int32)
+        Right: IBinaryOperatorExpression (BinaryOperationKind.IntegerAdd) (OperationKind.BinaryOperatorExpression, Type: System.Int32)
+            Left: ILocalReferenceExpression: i (OperationKind.LocalReferenceExpression, Type: System.Int32)
+            Right: ILiteralExpression (Text: 1) (OperationKind.LiteralExpression, Type: System.Int32, Constant: 1)
+  IBlockStatement (0 statements) (OperationKind.BlockStatement)
+";
+            VerifyOperationTreeForTest<ForStatementSyntax>(source, expectedOperationTree);
+        }
+
+        [Fact, WorkItem(17602, "https://github.com/dotnet/roslyn/issues/17602")]
+        public void IForLoopStatement_DecreasingForIterator()
```

Comment:
Not really sure what interesting property of for loops this is testing? Maybe Aleksey will disagree with me, but I don't see anything that should cause a real difference in the bound trees between this and a standard for loop